### ISSUE TITLE
Some tests not skipped in incremental mode.

### ISSUE THREAD

**Issue by mkizka** [stryker-mutator/stryker-js#3909]
**Summary**
Some tests run each time in incremental mode. I expect the tests to be skipped on the second run.

The tests like:

```js
test.each`
  x    | y    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${2} | ${4}
`("add($x, $y) = $expected", ({ x, y, expected }) => {
  expect(add(x, y)).toBe(expected);
});

```

When such a test exists, stryker shows `Tests: 1 files changed (+2 -0)` even though I haven't changed any tests.

The cause seems to be that each test has the same start position. [This code](https://github.com/stryker-mutator/stryker-js/blob/cef6abd978760234708ca242934dc4da31e1fec5/packages/core/src/mutants/incremental-differ.ts#L534) removes tests with the same start position.

Reproduction steps:
```
$ git clone https://github.com/mkizka/stryker-js-jest-each
$ cd stryker-js-jest-each
$ yarn
$ yarn stryker run --incremental
$ yarn stryker run --incremental
```

Expected behavior:
All tests are skipped the second time.

Actual behavior:
Some tests run the second time.

I'm using google translate, so I'm sorry if there are strange parts in the sentences.

**Stryker config**
```json
{
  "$schema": "./node_modules/@stryker-mutator/core/schema/stryker-schema.json",
  "_comment": "This config was generated using 'stryker init'. Please take a look at: https://stryker-mutator.io/docs/stryker-js/configuration/ for more information.",
  "packageManager": "yarn",
  "reporters": ["html", "clear-text", "progress"],
  "testRunner": "jest",
  "testRunner_comment": "More information about the jest plugin can be found here: https://stryker-mutator.io/docs/stryker-js/jest-runner",
  "coverageAnalysis": "perTest"
}
```

**Stryker environment**
```
+-- @stryker-mutator/core@6.3.0
+-- @stryker-mutator/jest-runner@6.3.0
```
```
+-- jest@29.3.1
```

**Test runner environment**
```shell
$ jest
```

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | v18.12.1
| npm              | 9.2.0
| Operating System | Ubuntu(WSL2)

**Add stryker.log**
[stryker.log](https://github.com/stryker-mutator/stryker-js/files/10253316/stryker.log) (second run in reproduction steps)

**Comment by nicojs** [comment]
Hi @mkizka, thanks for opening this issue with detailed reproduction steps!

This is a bug, but not in the line of code you're referring to the problem [is here](https://github.com/stryker-mutator/stryker-js/blob/cef6abd978760234708ca242934dc4da31e1fec5/packages/core/src/mutants/incremental-differ.ts#L549). I forgot to account for more than 1 test starting on the same line _when they are the last test in the file 🤦‍♀️_. 

A current workaround is to add a final test under the tests with the same start position:

```diff
const { add } = require("../add");

test.each`
  x    | y    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${2} | ${4}
`("add($x, $y) = $expected", ({ x, y, expected }) => {
  expect(add(x, y)).toBe(expected);
});

+ test('one final test', () => {});
```

**Comment by mkizka** [comment]
Thank you for your answer and quick fix!

**Comment by nicojs** [comment]
Sure thing ☺. I had some time, so you reported it just at the right moment 🤗

I also fixed the bug and released the fix in version 6.3.1
